File: 백준/Silver/8911. 거북이/거북이.js
let input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const n = +input.shift();

function solution(){
	for(let i = 0; i<n; i++){
		let move = input[i];
		let d = 0; //방향
		let up = 0, left = 0, down = 0, right = 0;
		let a = 0, b = 0; //가로 세로
		
		let maxa = 0, mina = 0;
		let maxb = 0, minb = 0;
		for(let k = 0; k < move.length; k++){
			if(move[k] == 'F'){
				if(d == 0) a ++;
				else if(d == 1) b++;
				else if(d == 2) a--;
				else b--;
			}
			else if(move[k] == 'B'){
				if(d == 0) a--;
				else if(d == 1) b--;
				else if(d == 2) a++;
				else b++;
			}
			else if(move[k] == 'L') d  = (d + 1) % 4;
			else if(move[k] == 'R'){
				d--;
				if(d < 0) d += 4;
			}
			
			if(a > 0) maxa = Math.max(a,maxa);
			else if(a < 0) mina = Math.min(mina,a);
			
			if(b > 0) maxb = Math.max(b,maxb);
			else if(b < 0) minb = Math.min(minb,b);
		}
		
		console.log((maxa - mina) * (maxb - minb));
	}
}

solution();

File: 백준/Silver/8911. 거북이/거북이.java
import java.io.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		for(int i = 0; i < n; i++){
			String s = br.readLine();
			char[] arr = s.toCharArray();
			System.out.println(sol(arr));
		}
	}
	
	public static int[] dx = {-1,0,1,0};
	public static int[] dy = {0,1,0,-1};
	
	public static int sol(char[] arr){
		int loc = 0;
		int a = 0; int b = 0;
		int maxA = 0; int maxB = 0;
		int minA = 0; int minB = 0;
		
		for(int j = 0; j < arr.length; j++){
			if(arr[j] == 'F'){
				a += dx[loc];	
				b += dy[loc];	
			} else if(arr[j] == 'B'){
				a -= dx[loc];	
				b -= dy[loc];	
			} else if(arr[j] == 'R'){
				loc++;
				loc %= 4;
			} else{
				loc--;
				if(loc < 0) loc = 3;
			}
			maxA = Math.max(maxA, a);
			maxB = Math.max(maxB, b);
			minA = Math.min(minA, a);
			minB = Math.min(minB, b);
		}
		
		return (maxA - minA) * (maxB - minB);
	}
}